chess/xboard.py

```diff
@@ -355,21 +355,42 @@ def hard(self, async_callback=None):
         """
         self.pondering(True)
 
-    def post(self, async_callback=None):
+    def set_post(self, flag, async_callback=None):
         """
-        Command used to tell the engine to output it's analysis
+        Command used to tell the engine whether to output it's analysis or not.
+
+        :param flag: True or False to set post on or off.
 
         :return: Nothing
         """
         def command():
             with self.semaphore:
-                self.send_line("post")
+                if flag == True:
+                    self.send_line("post")
+                else:
+                    self.send_line("nopost")
 
                 if self.terminated.is_set():
                     raise EngineTerminatedException()
 
         return self._queue_command(command, async_callback)
 
+    def post(self, async_callback=None):
+        """
+        Command used to tell the engine to output it's analysis.
+
+        :return: Nothing
+        """
+        self.set_post(True)
```

Comment:
This should probably be `return self.set_post(True)`, so that it can return a `Future` depending on `async_callback`.